### ISSUE TITLE
`azurerm_machine_learning_workspace` is rebuilt when `public_access_behind_virtual_network_enabled` is changed; inconsistent with portal behavior

### ISSUE THREAD

**Issue by SudoSpartanDan** [hashicorp/terraform-provider-azurerm#17170]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.0.4

### AzureRM Provider Version

Latest

### Affected Resource(s)/Data Source(s)

azurerm_machine_learning_workspace

### Terraform Configuration Files

```hcl
resource "azurerm_machine_learning_workspace" "main" {
  name                    = XXXXX
  friendly_name           = XXXXX
  location                = XXXXX
  resource_group_name     = XXXXX
  application_insights_id = XXXXX
  key_vault_id            = XXXXX
  storage_account_id      = XXXXX
  container_registry_id   = XXXXX

  identity {
    type = "SystemAssigned"
  }

  public_access_behind_virtual_network_enabled = false
  image_build_compute_name                     = "image-builder"
}
```

### Debug Output/Panic Output

```shell
The workspace is re-built when this setting is changed
```

### Expected Behaviour

In the Azure Portal UI, this setting can be toggled without rebuilding the resource

### Actual Behaviour

When this resource is changed, it wants to re-create the whole workspace

### References

https://github.com/hashicorp/terraform-provider-azurerm/blob/57cecc38dfd3e474723514c87756eb3d96f2a7ae/internal/services/machinelearning/machine_learning_workspace_resource.go#L115

**Comment by SudoSpartanDan** [comment]
I am no longer working on this technology and cannot test the change. It's been over a year since I last submitted this so not sure how much help I can be to test it.

**Comment by rcskosir** [comment]
Thanks for the update. It looks like this has been resolved as of #18279. We’re going to close this issue for now, but if you or anyone who stumbles upon this issue runs into this again with the latest Terraform and Provider versions, please feel free to [open a new issue](https://github.com/hashicorp/terraform-provider-azurerm/issues/new/choose).
